### PR TITLE
update dependies.

### PR DESCRIPTION
**Pull request by morvencao**
update dependies:

- controller-runtime: v0.9.0
- k8s-api: v1.21.1

also add `testdata` dir for test data, actually they contain empty files.

Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

**Review by songleo — APPROVED** [review]

File: controllers/multiclusterobservability/multiclusterobservability_controller_test.go [hunk 1]
```diff
@@ -242,15 +242,15 @@ func TestMultiClusterMonitoringCRUpdate(t *testing.T) {
 	}
 	//provide a non-existence path to bypass the rendering
 	//cannot convert unstructured.Unstructured into v1.Service in fake client
-	os.Setenv("TEMPLATES_PATH", path.Join(wd, "../../../tests/manifests"))
+	os.Setenv("TEMPLATES_PATH", path.Join(wd, "testdata/manifests"))
```
Comment (clyang82) [inline controllers/multiclusterobservability/multiclusterobservability_controller_test.go:245]: Why we cannot use `../../../tests` as template_path?

File: controllers/multiclusterobservability/multiclusterobservability_controller_test.go [hunk 2]
```diff
@@ -298,6 +298,11 @@ func TestMultiClusterMonitoringCRUpdate(t *testing.T) {
 	// Create a ReconcileMemcached object with the scheme and fake client.
 	r := &MultiClusterObservabilityReconciler{Client: cl, Scheme: s, OcpClient: ocpClient, CrdClient: crdClient}
 	config.SetMonitoringCRName(name)
+
+	err = cl.Create(context.TODO(), mco)
```
Comment (clyang82) [inline controllers/multiclusterobservability/multiclusterobservability_controller_test.go:302]: What is different between init object with call `create` explicitly?

Comment (morvencao) [reply controllers/multiclusterobservability/multiclusterobservability_controller_test.go:302]: This change doesn't matter actually based on my experiments.
The one matters is the MCO CR delete option, currently the MCO CR deletion will be returned soon, but the MCO CR is actually still there because the finalizer is not cleaned.

I tried to add a reconcile loop after the deletion operator, but the reconcile loop hang somewhere. I guess it may be related to the status update goroutine, just my guess.

Currently, the most simple fix is to delete the finalizer manually after the MCO CR deletion operaton, then the MCO CR can be successfully deleted.

File: controllers/multiclusterobservability/multiclusterobservability_controller_test.go (hunk 1 above)
Comment (morvencao) [reply controllers/multiclusterobservability/multiclusterobservability_controller_test.go:245]: Looks like we have to test this with directory exists, otherwise we will see lots of warning:
```
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/observatorium, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/config, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/proxy, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/grafana, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:07 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/thanos, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/observatorium, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/config, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/proxy, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/grafana, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:09 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/thanos, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/observatorium, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/config, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/proxy, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/grafana, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:10 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/thanos, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/observatorium, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/observatorium' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/config, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/config' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/proxy, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/proxy' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/grafana, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/grafana' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/alertmanager' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
2021/06/15 08:36:11 Cannot find this path /go/src/github.com/open-cluster-management/tests/manifests/base/thanos, absolute path error in '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : evalsymlink failure on '/go/src/github.com/open-cluster-management/tests/manifests/base/thanos' : lstat /go/src/github.com/open-cluster-management/tests: no such file or directory
```

**Comment by morvencao** [comment]
@clyang82 Another look at this?

File: controllers/multiclusterobservability/multiclusterobservability_controller.go [hunk 3]
```diff
@@ -238,7 +238,9 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 	}
 
 	//update status
-	requeueStatusUpdate <- struct{}{}
+	if updateStatusIsRunnning {
+		requeueStatusUpdate <- struct{}{}
+	}
```
Comment (morvencao) [inline controllers/multiclusterobservability/multiclusterobservability_controller.go:243]: make sure the status update channel is running before send signal, otherwise the reconcile will hang here.
